src/pal/src/misc/miscpalapi.cpp

```diff
@@ -126,11 +115,94 @@ PAL_GetPALDirectoryW( OUT LPWSTR lpDirectoryName, IN UINT cchDirectoryName )
     /* Error path, should not be executed. */
     SetLastError( ERROR_INTERNAL_ERROR );
 EXIT:    
-    LOGEXIT( "PAL_GetPALDirectoryW returns BOOL %d.\n", bRet);
     PERF_EXIT(PAL_GetPALDirectoryW);
     return bRet;
 }
 
+BOOL
+PAL_GetPALDirectoryA(PathCharString& lpDirectoryName)
+{
+    BOOL bRet;
+    PathWCharString directory;
+
+    PERF_ENTRY(PAL_GetPALDirectoryA);
+
+    bRet = PAL_GetPALDirectoryW(directory);
+
+    if (bRet) {
+        
+        int length = WideCharToMultiByte(CP_ACP, 0, directory.GetString(), -1, NULL, 0, NULL, 0);
+        LPSTR DirectoryName = lpDirectoryName.OpenStringBuffer(length);
+        if (NULL == DirectoryName)
+        {
+            SetLastError( ERROR_INSUFFICIENT_BUFFER );
+            bRet = FALSE;
+        }
+        
+        length = WideCharToMultiByte(CP_ACP, 0, directory.GetString(), -1, DirectoryName, length, NULL, 0);
+        lpDirectoryName.CloseBuffer(length - 1);
+
+        if (0 == length)
```

Comment:
bRet is TRUE here, so you can simplify the "if -else" to just if for the FALSE case.